nixos/tests/gnome3/flashback.nix

```diff
@@ -0,0 +1,95 @@
+# GNOME Flashback Terminal tests for various custom sessions.
+
+{ system ? builtins.currentSystem,
+  config ? {},
+  pkgs ? import ../../.. { inherit system config; }
+}:
+
+with import ../../lib/testing.nix { inherit system pkgs; };
+with pkgs.lib;
+
+let
+
+  baseConfig = {
+    imports = [ ../common/user-account.nix ];
+
+    services.xserver.enable = true;
+
+    services.xserver.displayManager.gdm.enable = false;
```

Comment:
Could you add a comment why this is done? I do not think `git blame` will carry to the original file

Reply:
Hmm, it seems in the `gnome3-xorg` test, and in the past gnome3 module enabled gdm and then it got removed and this got added. And I've copied it here.

Actual issue with why we don't test with gdm is https://github.com/NixOS/nixpkgs/issues/66443.

Reply:
Yeah, link there  should be enough.

Reply:
GDM doesn't have any way to specify a default session, and I'm guessing it will happily launch wayland sessions even though GDM itself runs X11. Think I switched the test to LightDM when adding wayland support since I didn't manage to get it launch the correct session.

We'd probably have to do something with AccountsService to get it working.

Reply:
Huh, that's weird, GDM X11 does indeed launch an X11 session, never mind me.

Reply:
Yeah, I wasn't sure why I was seeing `error: unable to open display ":0.0"` when using `xwininfo`. Only thing I thought of was it using wayland? But it was explicitly disabled.

Reply:
@hedning You're actually correct, setting xserver `default` will have no affects on gdm.
So we do need to set it with accountsservice over dbus.